pkg/utils/kubernetes/health/checker/checker.go

```diff
@@ -116,8 +116,8 @@ func (h *HealthChecker) checkEtcds(condition gardencorev1beta1.Condition, object
 				codes   []gardencorev1beta1.ErrorCode
 			)
 
-			if lastError := object.Status.LastError; lastError != nil {
-				message = fmt.Sprintf("%s (%s)", message, *lastError)
+			if len(object.Status.LastErrors) != 0 {
+				message = fmt.Sprintf("%s (%+v)", message, object.Status.LastErrors)
```

Comment:
If a shoot was not yet reconciled, it still contains the old `.status.lastError`, right? Hence, if this contains content, wouldn't we report etcd as "healthy" because we now only check for `.status.lastErrors`?

Reply:
`.status.lastError` has been removed from the API, so it's no longer available in the status. Nevertheless, druid reconciles the Etcd resource status periodically (every 15 seconds as configured). So it will eventually update the status with `.status.lastErrors` and the gardener condition check will eventually reflect that.